spec/lib/tasks/scripts/block_level_sync_spec.rb

```diff
@@ -0,0 +1,105 @@
+require "rails_helper"
+require "tasks/scripts/block_level_sync"
+
+RSpec.describe BlockLevelSync do
+  describe ".enable" do
+    it "enables the specified user_ids" do
+      enabled_users = create_list(:user, 5)
+      not_enabled_users = create_list(:user, 5)
+
+      BlockLevelSync.enable(enabled_users)
+
+      enabled_users.each do |u|
+        expect(u.block_level_sync?).to eq(true)
+      end
+
+      not_enabled_users.each do |u|
+        expect(u.block_level_sync?).to eq(false)
+      end
+    end
+
+    it "touches the facilities for the user" do
+      enabled_users = create_list(:user, 5)
+      enable_time = Time.new("2018-1-1")
+
+      Timecop.freeze(enable_time) do
+        BlockLevelSync.enable(enabled_users)
+      end
+
+      enabled_users.each do |u|
+        u.facility_group.facilities.pluck(:updated_at).each do |t|
+          expect(t.to_i).to eq(enable_time.to_i)
+        end
+      end
+    end
+  end
+
+  describe ".disable" do
+    it "disables the specified user_ids" do
+      enabled_users = create_list(:user, 5)
+      not_enabled_users = create_list(:user, 5)
+
+      BlockLevelSync.enable(enabled_users)
+
+      enabled_users.each do |u|
+        expect(u.block_level_sync?).to eq(true)
+      end
+
+      BlockLevelSync.disable(enabled_users)
+
+      enabled_users.each do |u|
+        expect(u.block_level_sync?).to eq(false)
+      end
+
+      not_enabled_users.each do |u|
+        expect(u.block_level_sync?).to eq(false)
+      end
+    end
+  end
+
+  describe ".bump" do
+    it "enables a percentage of users randomly" do
+      users = create_list(:user, 20)
+      allow(Reports::RegionCacheWarmer).to receive(:call).and_return(true)
+
+      BlockLevelSync.bump(95)
+
+      expect(users.map(&:block_level_sync?).count(&:itself)).to be_between(17, 21)
+    end
+
+    it "increases the percentage" do
+      users = create_list(:user, 20)
+      allow(Reports::RegionCacheWarmer).to receive(:call).and_return(true)
+
+      BlockLevelSync.bump(95)
+
+      expect(users.map(&:block_level_sync?).count(&:itself)).to be_between(17, 21)
```

Comment:
Why would there be 21 users if we're only creating 20 above? 

PS is there any way to get a more precise assertion? I'm guessing we're using a range because there's some randomness. Can we instead (or also) query the the Flipper percentage for the specific feature flag?

Reply:
I added another clause to check the actual percentage value here https://github.com/simpledotorg/simple-server/pull/1916/commits/8ba088d7ae6b1189ea0194942d7aecf389817025 and modified the specs accordingly.